xmsannotator/R/compute_scores.R

```diff
@@ -1,35 +1,101 @@
+#' Assign abundance ratio of the second most abundant isotope/isotopologue for each entry in a table
+#'
+#' @param isotopes A table with annotated peaks filtered by adducts
+#' @return The input table with extra column containing the abundance ratio of the second most abundant isotope/isotopologue
+#'
+#' @import dplyr
+assign_isotope_abundances <- function(isotopes) {
+  molecules <- distinct(isotopes, molecular_formula)
+  molecules <- mutate(molecules, abundance_ratio = map_dbl(molecular_formula, compute_abundance_ratio))
+  isotopes <- inner_join(isotopes, molecules, by = "molecular_formula")
+}
+
+#' Get abundance ratio of the second most abundant isotope/isotopologue
+#'
+#' @param formula A string containing molecular or empirical formula of a compound
+#'
+#' @return Relative abundance ratio of the second most abundant isotope/isotopologue
+#'
+#' @importFrom Rdisop getMolecule
+compute_abundance_ratio <- function(formula) {
+  molecule <- getMolecule(formula)
+  abundance_ratios <- sort(molecule$isotopes[[1]][2,], decreasing = TRUE)
+  sec_most_abundant <- abundance_ratios[2]
+}
+
+#' Match isotopes from peak table to annotated peaks
+#'
+#' @param ... A single isotope passed as a list of columns
+#' @param intensity_deviation_tolerance A number. A threshold by which an intensity ratio of isotope
+#' @param peaks A peak table - `isp_masses_mz_data` from the original tool's multilevelannotation step 1

```

Comment:
The sentence for the first parameter contains some errors.

The documentation of the parameters could be improved - for example which columns of the peak table are used and expected, what are the data formats etc.

Reply:
Thanks, I actually didn't complete the description of this parameter and forgot about it. See improved documentation in 0abaad14dba6779f8e743f3daec8153d3eb75ea8